### ISSUE TITLE
Import assertions not allowed in nodenext

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#46372]
I tried using import assertions while using `"module": "nodenext"`.

```ts
// @module: nodenext

import json from "some-module" assert { type: "json" };
//                             ~~~~~~~~~~~~~~~~~~~~~~~
// error:
// Import assertions are only supported when the '--module' option is set to 'esnext'.
```

**Expected**: no error

**Actual**:

```
Import assertions are only supported when the '--module' option is set to 'esnext'.
```
